rpc/resource/internal/string.go

```diff
@@ -0,0 +1,45 @@
+package internal
+
+import "strings"
+
+const (
+	// Delimiter of Resource Reference according to Atlas Reference format
+	Delimiter = "/"
+)
+
+// BuildString builds string id according to Atlas Reference format:
+//  <application_name>/<resource_type>/<resource_id>
+func BuildString(aname, rtype, rid string) string {
+	var l []string
+
+	if v := strings.TrimSpace(aname); v != "" {
+		l = append(l, v)
+	}
+	if v := strings.TrimSpace(rtype); v != "" {
+		l = append(l, v)
+	}
+	if v := strings.TrimSpace(rid); v != "" {
+		l = append(l, v)
+	}
+
+	return strings.Join(l, Delimiter)
+}
+
+// ParseString parses id according to Atlas Reference format:
+//	<application_name>/<resource_type>/<resource_id>
+// All leading and trailing Delimiter are removed.
+// The resource_id is parsed first, then resource type and last application name.
+// The id "/a/b/c/" will be converted to "a/b/c" and returned as (a, b, c).
+// The id "b/c/" will be converted to "b/c" and returned as ("", b, c).
+func ParseString(id string) (aname, rtype, rid string) {
```

Comment:
it might be useful to publicly export this, in cases where people are storing references as a string instead of our custom type

Reply:
it might be, but let's keep it private for now. It is easy to make functions public than private.